### PR TITLE
config: Respect --yaml-config option, if it is set

### PR DESCRIPTION
**Pull request by nuclearcat**
Fixes https://github.com/kernelci/kernelci-core/issues/2120

As discussed, if --yaml-config set - it will load config files from specified directory, which allows us to make more complex configuration with multiple files.

### CODE REVIEW

**Review by gctucker — COMMENTED** [review]
Are you sure this works?  I would expect the default files in `/etc/kernelci/core` to be loaded by default so `get_yaml_configs()` should never be empty.

Probably the best way to actually solve this is to drop the hard-coded `config/pipeline.yaml` path now and just pass it as a command line argument in `docker-compose.yaml`.

**Comment by gctucker** [comment]
Ah I think it kind of works because `Options.get_yaml_configs()` is kind of misleading, it doesn't really tell the full list of YAML files being loaded but the value passed on the command line.

The `kernelci.config.get_config_paths()` function actually returns the list of YAML files loaded by the configuration, including system defaults in `/etc/kernelci/core` and `./config/core`.

Alright so that's why this PR works.  Still it would be better to avoid duplicating the call to the load method with a variable I think - but that's implementation (following the DRY principle).

File: src/regression_tracker.py [hunk 1]
```diff
@@ -106,6 +106,9 @@ def __call__(self, configs, args):
 
 if __name__ == '__main__':
     opts = parse_opts('regression_tracker', globals())
-    configs = kernelci.config.load('config/pipeline.yaml')
+    if len(opts.get_yaml_configs()) > 0:
+        configs = kernelci.config.load(opts.get_yaml_configs())
+    else:
+        configs = kernelci.config.load('config/pipeline.yaml')
```
[the remark this answers is not in the document]
Comment (nuclearcat) [reply src/regression_tracker.py]: Updated

**Review by gctucker — APPROVED** [review]
Tested OK on staging.

**Review by gctucker — COMMENTED** [review]
The git commit prefix is not consistent with the history though:
```
config: Respect --yaml-config option, if it is set
```
It should be:
```
src: Respect --yaml-config option, if it is set
```

**Pull request merged**